lib/Backend/arm64/LowerMD.cpp

```diff
@@ -936,32 +898,29 @@ LowererMD::GenerateStackAllocation(IR::Instr *instr, uint32 allocSize, uint32 pr
     //__chkStk is a leaf function and hence alignment is not required.
 
     // Generate _chkstk call
-    // LDIMM RegR4, stackSize/4         //input:  r4 = the number of WORDS (word = 4 bytes) to allocate,
-    // LDIMM RegR12, HelperCRT_chkstk
-    // BLX RegR12
-    // SUB SP, SP, RegR4                //output: r4 = total number of BYTES probed/allocated.
+    // LDIMM RegR15, stackSize/16
+    // LDIMM RegR17, HelperCRT_chkstk
+    // BLX RegR17
+    // SUB SP, SP, x15, lsl #4
 
-    // ARM64_WORKITEM: This needs to be fixed up to remove R12 dependency
-    __debugbreak();
-        
-    //chkstk expects the stacksize argument in R4 register
-    IR::RegOpnd *r4Opnd = IR::RegOpnd::New(nullptr, SP_ALLOC_SCRATCH_REG, TyMachReg, this->m_func);
+    //chkstk expects the stacksize argument in R15 register
+    IR::RegOpnd *spAllocOpnd = IR::RegOpnd::New(nullptr, SP_ALLOC_SCRATCH_REG, TyMachReg, this->m_func);
     IR::RegOpnd *targetOpnd = IR::RegOpnd::New(nullptr, SCRATCH_REG, TyMachReg, this->m_func);
 
-    IR::IntConstOpnd *  stackSizeOpnd   = IR::IntConstOpnd::New((allocSize/MachPtr), TyMachReg, this->m_func, true);
+    IR::IntConstOpnd *  stackSizeOpnd   = IR::IntConstOpnd::New((allocSize / MachStackAlignment), TyMachReg, this->m_func, true);
 
-    IR::Instr *movInstr = IR::Instr::New(Js::OpCode::LDIMM, r4Opnd, stackSizeOpnd, this->m_func);
+    IR::Instr *movInstr = IR::Instr::New(Js::OpCode::LDIMM, spAllocOpnd, stackSizeOpnd, this->m_func);
     instr->InsertBefore(movInstr);
```

Comment:
Is it ok to use LDIMM in the prolog, which might get expanded to multiple instructions? (Just checking, I don't know for sure if we're more restricted here.)

Reply:
Is there some other instruction which can achieve the same?

Reply:
You can use MOVZ directly if you know the immediate is less than 64k. But again, if it's correct to use LDIMM and allow for expansion, feel free to keep it as-is. For some reason had it in my head that there were some limitations in this situation, but I can't point to how I suspect this. :)

Reply:
Thanks! for now will keep it this way.